src/ssl.c

```diff
@@ -15537,6 +15645,88 @@ char* wolfSSL_ASN1_TIME_to_string(WOLFSSL_ASN1_TIME* t, char* buf, int len)
 }
 #endif /* WOLFSSL_MYSQL_COMPATIBLE */
 
+#if defined(OPENSSL_EXTRA) && !defined(NO_ASN_TIME)
+
+WOLFSSL_ASN1_TIME* wolfSSL_ASN1_TIME_adj(WOLFSSL_ASN1_TIME *s, time_t t,
+                                    int offset_day, long offset_sec)
+{
+    const int sec_per_day = 24*60*60;
+    const int utc_sz = 14;
+    const int general_sz = 16;
+    struct tm* ts = NULL;
+    time_t t_adj = 0;
+    time_t offset_day_sec = 0;
+
+    WOLFSSL_ENTER("wolfSSL_ASN1_TIME_adj");
+
+    if (s == NULL){
+        s = (WOLFSSL_ASN1_TIME*)XMALLOC(sizeof(WOLFSSL_ASN1_TIME), NULL,
```

Comment:
Thanks, yes we should follow the same usage as OpenSSL.  Documentation here will be very helpful for end users.